navigationrail/navigationrail/src/main/java/com/microsoft/device/dualscreen/navigationrail/NavigationRailView.kt

```diff
@@ -0,0 +1,522 @@
+            return
+        }
+
+        val childMenuTop = calculateMenuTopMargin()
+        val defaultChildHeight = getChildAt(0).measuredHeight
+        val availableHeightOnTopScreen = topScreenHeight - childMenuTop
+        val skipAnimation = shouldSkipAnimation(this)
+
+        var btnMargin = 0
+
+        // Calculate a negative button margin if there is not enough space on the top
+        if (defaultChildHeight * topBtnCount > availableHeightOnTopScreen) {
+            btnMargin =
+                (availableHeightOnTopScreen - defaultChildHeight * topBtnCount) / topBtnCount
+        }
+
+        val topStartingPosition =
+            availableHeightOnTopScreen - topBtnCount * (defaultChildHeight + btnMargin)
+
+        for (i in 0 until topBtnCount) {
+            val childTop = topStartingPosition + i * (defaultChildHeight + btnMargin)
+            val child = getChildAt(i)
+            child.layout(child.left, 0, child.right, 0 + defaultChildHeight)
+            setButtonPosition(child, childTop, skipAnimation)
+        }
+
+        val availableHeightOnBottomScreen = appWindowPosition.bottom - hingePosition.bottom
+        val bottomStartingPosition = this.height - availableHeightOnBottomScreen
+
+        // Calculate a negative button margin if there is not enough space on the bottom
+        if (defaultChildHeight * bottomBtnCount > availableHeightOnBottomScreen) {
+            btnMargin =
+                (availableHeightOnBottomScreen - defaultChildHeight * bottomBtnCount) / bottomBtnCount
+        }
+
+        for (i in topBtnCount until buttonsCount) {
+            val childTop = bottomStartingPosition +
+                (i - topBtnCount) * (defaultChildHeight + btnMargin)
+            val child = getChildAt(i)
+            child.layout(child.left, 0, child.right, 0 + defaultChildHeight)
+            setButtonPosition(child, childTop, skipAnimation)
+        }
+    }
+
+    /**
+     * Sets the position inside the [NavigationRailMenuView] and triggers the translation animations.
+     */
+    private fun setButtonPosition(
+        child: View,
+        childTop: Int,
+        skipAnimation: Boolean
+    ) {
+        if (skipAnimation || !useAnimation || Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
+            child.translationY = childTop.toFloat()
+        } else {
+            child.animate()
+                .setInterpolator(animationInterpolator)
+                .translationY(childTop.toFloat())
+        }
+    }
+
+    /**
+     * Returns the margin of the [NavigationRailMenuView] from the top of it's parent.
+     * This is useful when the [NavigationRailView] contains a header view.
+     */
+    private fun calculateMenuTopMargin(): Int {
+        val topMargin =
+            resources.getDimensionPixelSize(com.google.android.material.R.dimen.mtrl_navigation_rail_margin)
+        return headerView?.let {
+            it.bottom + topMargin
+        } ?: kotlin.run { 0 }
```

Comment:
You can return 0 without to invoke "run" function

Reply:
Fixed